pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/io/ConnectorUtils.java

```diff
@@ -53,6 +59,8 @@ public class ConnectorUtils {
 
     private static final String PULSAR_IO_SERVICE_NAME = "pulsar-io.yaml";
 
+    private static final Executor cachedThreadPool = Executors.newCachedThreadPool();
```

Comment:
We have to control the life cycle of this ThreadPool
leaving a static ThreadPool is generally a not a good practice.
In this case we can create the ThreadPool and dispose it in searchForConnectors.

Also I would use a fixed size threadpool, otherwise we have to control on the parallelism

Reply:
we should also provide a meaningful name to the threads